providers/implementations/signature/dsa_sig.c

```diff
@@ -172,8 +172,8 @@ static int dsa_setup_md(PROV_DSA_CTX *ctx,
             && ossl_DER_w_algorithmIdentifier_DSA_with_MD(&pkt, -1, ctx->dsa,
                                                           md_nid)
             && WPACKET_finish(&pkt)) {
-            WPACKET_get_total_written(&pkt, &ctx->aid_len);
-            ctx->aid = WPACKET_get_curr(&pkt);
+            if(WPACKET_get_total_written(&pkt, &ctx->aid_len))
+                ctx->aid = WPACKET_get_curr(&pkt);
```

Comment:
This really really cannot fail. I do not think it makes sense to add the check.

Reply:
To put this another way, the assumption of every WPACKET_ related API is that the pkt pointer must be valid.
As it does not check this the only other possible failure is if the second param is NULL.
If that behaviour was  to change internally then your suggested change would be needed. 
So it is not currently needed, but it is future proof.
Maybe, this is an example of where the API may have been better to just return the value rather than an error.

Reply:
What should we do about this issue?